processor/src/main/java/org/mapstruct/ap/internal/model/MapMappingMethod.java

```diff
@@ -319,6 +319,9 @@ public boolean equals(Object obj) {
         }
 
         for ( int i = 0; i < getSourceParameters().size(); i++ ) {
+            if ( !getSourceParameters().get( i ).getType().equals( other.getSourceParameters().get( i ).getType() ) ) {
+                return false;
+            }
             if ( !getSourceParameters().get( i ).getType().getTypeParameters().get( 0 )
```

Comment:
You did not touch this.. but looking upon this code, line 325 cannot be correct. For `Iterable` types we decide to check all type parameters. Here only the first..? Feels not logical for maps. Can you copy the sam code here as for the `Iterable` method?

So:
```java
            List<Type> thisTypeParameters = getSourceParameters().get( i ).getType().getTypeParameters();
            List<Type> otherTypeParameters = other.getSourceParameters().get( i ).getType().getTypeParameters();

            if ( !thisTypeParameters.equals( otherTypeParameters ) ) {
                return false;
            }
```